test/jest.setup.js

```diff
@@ -1,6 +1,26 @@
 // This filename is not magic; it is referenced by Jest's setupFilesAfterEnv
 
 import { setDefaultOptions } from 'expect-puppeteer';
+import { CLIENT_URL } from './src/util';
 
 // Set the default action timeout to something greater than 500ms
 setDefaultOptions({ timeout: 10000 });
+
+// this is not redundant
+// it allows tests to be logged in already before doing setup in their own beforeAlls
+// eslint-disable-next-line jest/require-top-level-describe
+beforeAll(async () => {
+  await Promise.all([
+    page.goto(CLIENT_URL),
+    page.waitForNavigation({ waitUntil: 'networkidle0' }),
+  ]);
+});
+
+// eslint-disable-next-line jest/require-top-level-describe
+beforeEach(async () => {
+  await jestPuppeteer.resetBrowser();
+  await Promise.all([
+    page.goto(CLIENT_URL),
+    page.waitForNavigation({ waitUntil: 'networkidle0' }),
+  ]);
+});
```

Comment:
I'm confused as to the reason the above `beforeAll` is needed if this is also here? Wouldn't this cover all of the necessary functionality of the `beforeAll`?

Reply:
This is interestingly arcane, and I intended the documentation on the `beforeAll` to explain it. I will rework that more thoroughly.
Basically jest runs `before` and `after` functions in this order:
* global `beforeAll`, like this one
* `beforeAll` in the test file
* global `beforeEach`
* `beforeEach` in the test file
* the test code
* `afterEach` and `afterAll`, in reverse of the `before` functions
Tests should be able to assume that before each test method, the browser is a clean slate and is already sitting on the home page. The problem is that some tests also might want to set stuff up in a `beforeAll` (like creating a single dandiset so it can be searched for in a search test). According to the above run order, we also need a global `beforeAll` to start the test in the home page. Otherwise test writers will need to remember that `beforeEach` and `beforeAll` have different initial conditions, which is dumb.

Reply:
Makes sense. This should definitely be a code comment.